opentelemetry-api/src/opentelemetry/baggage/propagation/__init__.py

```diff
@@ -90,21 +104,33 @@ def inject(
         setter.set(carrier, self._BAGGAGE_HEADER_NAME, baggage_string)
 
     @property
-    def fields(self) -> typing.Set[str]:
+    def fields(self) -> Set[str]:
         """Returns a set with the fields set in `inject`."""
         return {self._BAGGAGE_HEADER_NAME}
 
 
-def _format_baggage(baggage_entries: typing.Mapping[str, object]) -> str:
-    return ",".join(
-        key + "=" + urllib.parse.quote_plus(str(value))
-        for key, value in baggage_entries.items()
-    )
+def _format_baggage(baggage_entries: Mapping[str, object]) -> str:
+
+    key_values = []
+
+    for key, value in baggage_entries.items():
+
+        if _key_regex.fullmatch(key) is None or (
+            _value_regex.fullmatch(str(value)) is None
+        ):
+            _logger.warning(
+                "Key %s and value %s have been discarded", key, value
+            )
+            continue
+
+        key_values.append(key + "=" + urllib.parse.quote(str(value)))
```

Comment:
Going off the issue, this `quote()` / `quote_plus()` thing is a separate issue? Would be good to document that in the PR/changelog